lib/processors/manifestEnricher.js

```diff
@@ -0,0 +1,313 @@
+const log = getLogger("builder:processors:manifestEnricher");
+
+const APP_DESCRIPTOR_V22 = new Version("1.21.0");
+
+/**
+ * Transforms i18n_en_US.properties to en_US
+ *
+ * @param {string} bundleName
+ * @param {string} defaultBundleName
+ */
+function getLocale(bundleName, defaultBundleName) {
+	log.verbose("Resource bundle detected on " + bundleName);
+	bundleName = path.basename(bundleName, ".properties");
+	defaultBundleName = path.basename(defaultBundleName, ".properties");
+	if (bundleName === defaultBundleName) {
+		return "";
+	} else {
+		return bundleName.replace(defaultBundleName + "_", "");
+	}
+}
+
+/**
+ * Determines if manifest contain handlebars placeholder templates
+ *
+ * Only in case of placeholders in manifest.json the bundle configured
+ * in sap.app/i18n is requested by sap/ui/core/Manifest.js
+ *
+ * @param {object} manifest The manifest
+ * @returns {boolean}
+ */
+function hasManifestTemplates(manifest) {
+	const rManifestTemplate = /\{\{([^}}]+)\}\}/g;
+	return JSON.stringify(manifest).match(rManifestTemplate);
+}
+
+/**
+ * Transforms "ui5://my/fancy/app/i18n/i18n.properties" to "my.fancy.app.i18n.i18n"
+ *
+ * @param {string} bundleUrl A bundle Url
+ * @returns {string}
+ */
+function getBundleNameByUrl(bundleUrl) {
+	return bundleUrl.split("ui5://")[1].replaceAll("/", ".").split(".properties")[0];
+}
+
+/**
+ * Determines if a bundle config should be enriched with supported locales
+ *
+ * @param {object} bundleConfig The bundle config
+ * @param {string} appId The app id
+ * @returns {boolean}
+ */
+function isAutofillRequired(bundleConfig, appId) {
+	const {supportedLocales, bundleUrl} = bundleConfig;
+	let {bundleName} = bundleConfig;
+
+	if (supportedLocales) {
+		return false;
+	}
+
+	if (bundleUrl && bundleUrl.startsWith("/")) {
+		log.verbose("manifest.json: bundleUrl '" + bundleUrl +
+			"' contains an absolute path, no supportedLocales are generated");
+		return false;
+	}
+
+	if (bundleUrl && bundleUrl.startsWith("..")) {
+		log.verbose("manifest.json: bundleUrl '" + bundleUrl +
+			"' contains a relative path, no supportedLocales are generated");
+		return false;
+	}
+
+	if (bundleUrl && bundleUrl.startsWith("ui5://")) {
+		bundleName = getBundleNameByUrl(bundleUrl);
+	}
+
+	if (bundleName && !bundleName.startsWith(appId)) {
+		const logMsg = bundleUrl ? "bundleUrl '" + bundleUrl : "bundleName '" + bundleName;
+		log.verbose("manifest.json: " + logMsg +
+			"' contains a path which is not part of the project, no supportedLocales are generated");
+		return false;
+	}
+
+	return true;
+}
+
+function getType(manifest) {
+	return manifest["sap.app"].type;
+}
+
+function isFallbackLocalePartOfSupportedLocales(fallbackLocale, supportedLocales) {
+	return supportedLocales.includes(fallbackLocale);
+}
+
+function hasBundleTerminologies(bundleConfig) {
+	return bundleConfig?.terminologies && Object.keys(bundleConfig.terminologies).length > 0;
+}
+
+function getTerminologieBundles(bundleConfig, bundleConfigs) {
```

Comment:
The name suggests that this should be a getter, but instead modifies `bundleConfigs`

Reply:
I've updated the PR to rather call those methods `process...`, and they also modify the manifest, but only in case supportedLocales are added.